File: Sliding Window/1. Best Time to Buy and Sell Stock/solution.rb
# @param {Integer[]} prices
# @return {Integer}
def max_profit(prices)
  min_left = 100000
  result = 0

  prices.each_with_index do |price, index|
    min_left = [min_left, price].min
    result = [result, price - min_left].max
  end

  result
end

File: Sliding Window/1. Best Time to Buy and Sell Stock/solution.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut min_left = 100000;
        let mut result = 0;
        for i in 0..prices.len() - 1 {
            min_left = prices[i].min(min_left);
            result = result.max(prices[i] - min_left);
        }
        result
    }
}
